lib/queue.rb

```diff
@@ -1,28 +1,33 @@
 class Queue
```

Comment:
This passes all the tests, but is _not_ implemented with a circular buffer.  I suggest you review the notes on the circular buffer just so you have a good understanding how a circular buffer works.